src/stm32/spi.c

```diff
@@ -41,10 +41,12 @@ DECL_CONSTANT_STR("BUS_PINS_spi4", "PE13,PE14,PE12");
 #define SPI_FUNCTION GPIO_FUNCTION(CONFIG_MACH_STM32F0 ? 0 : 5)
 
 static const struct spi_info spi_bus[] = {
+  { SPI1, GPIO('A', 6), GPIO('A', 7), GPIO('A', 5), SPI_FUNCTION },
+  { SPI1, GPIO('B', 4), GPIO('B', 5), GPIO('B', 3), SPI_FUNCTION },
+#ifdef SPI2
     { SPI2, GPIO('B', 14), GPIO('B', 15), GPIO('B', 13), SPI_FUNCTION },
-    { SPI1, GPIO('A', 6), GPIO('A', 7), GPIO('A', 5), SPI_FUNCTION },
     { SPI2, GPIO('C', 2), GPIO('C', 3), GPIO('B', 10), SPI_FUNCTION },
-    { SPI1, GPIO('B', 4), GPIO('B', 5), GPIO('B', 3), SPI_FUNCTION },
+#endif
```

Comment:
The `#ifdef` in this section must exactly match those in the DECL_ENUMERATION()/DECL_CONSTANT_STR() in the block above.  (That is, we need to ensure the enumeration assignments match the spi_bus array layout.)

Otherwise, the code looks good to me.
-Kevin

Reply:
I've updated the PR, also added a test for this platform.

Reply:
Thanks.  Unfortunately, the change does not look correct.  The DECL_ENUMERATION sets SPI2 to zero, but it is in the second slot of the array.  This needs to match.  (Also, FYI, the default SPI bus in the configs is the zero enumeration - and thus SPI2 is the default on several boards - so, we can't easily change SPI2 to a non-zero value on existing boards.)

-Kevin